src/test/streams_tests.cpp

```diff
@@ -101,8 +101,17 @@ BOOST_AUTO_TEST_CASE(streams_vector_reader)
     signed int d;
     BOOST_CHECK_THROW(reader >> d, std::ios_base::failure);
 
+    BOOST_CHECK(reader.empty());
+    BOOST_CHECK_THROW(reader.seek(-7), std::ios_base::failure);
+    BOOST_CHECK(reader.empty());
+
     // Read a 4 bytes as a signed int from the beginning of the buffer.
+    BOOST_CHECK_EQUAL(reader.size(), 0);
     reader.seek(-6);
+    BOOST_CHECK_EQUAL(reader.size(), 6);
+    BOOST_CHECK_THROW(reader.seek(-1), std::ios_base::failure);
+    BOOST_CHECK_EQUAL(reader.size(), 6);
+    BOOST_CHECK(!reader.empty());
```

Comment:
I don't think assertion is that useful after the one above. Correct behavior of `empty()` is sufficiently covered elsewhere in this test IMO.